Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h

```diff
@@ -249,6 +249,14 @@ class MediaPlayerPrivateGStreamer
     AbortableTaskQueue& sinkTaskQueue() { return m_sinkTaskQueue; }
 
     String codecForStreamId(const String& streamId);
+    bool shouldDownload() { return m_fillTimer.isActive(); }
+    void setQuirkState(const GStreamerQuirk* owner, GStreamerQuirkBase::GStreamerQuirkState&& state) { m_quirkStates.set(owner, WTF::makeUnique<GStreamerQuirkBase::GStreamerQuirkState>(WTFMove(state))); }
```

Comment:
The problem is that the HashMap documentation doesn't explain how to get a reference to the stored element, instead of the element itself (what does "the element itself" actually means? a copy of it? WTFMove() is nice to store things, but not to retrieve them. I don't want a copy, I want the real element (or a reference, or a pointer) that I can change and the changes stay inside the instance stored in the HashMap).

Can you point me to the documentation that explains how that can be done, or to a commit that has done that before?

That's why I thought that storing a pointer (specifically, an auto-destructible one, that is, unique_ptr) made more sense in this case, where I need to share GStreamerQuirkState between multiple clients.

Reply:
You're right. Forget about this.